src/Linkify.jsx

```diff
@@ -46,7 +48,14 @@ class Linkify extends React.Component {
         elements.push(string.substring(lastIndex, match.index));
       }
       // Shallow update values that specified the match
-      let props = {href: match.url, key: `parse${this.parseCounter}match${idx}`};
+      // Provide `preCompileURL` function in props, if you want to edit the url before
+      let props = {}
+      if(this.props.preCompileURL){
+        props = {href: this.props.preCompileURL(match.url), key: `parse${this.parseCounter}match${idx}`};
+      } else {
+        props = {href: match.url, key: `parse${this.parseCounter}match${idx}`};
+      }
```

Comment:
Here you're repeating yourself. I think something like this would be better:
```javascript
let url = match.url
if (this.props.preCompileURL) {
    url = this.props.preCompileURL(url)
}
let props = {href: url, key: `parse${this.parseCounter}match${idx}`};
```